### DISCUSSION TITLE
Is it possible to customize the "persona card on hover" card with custom attributes from AD?

### DISCUSSION THREAD

**Discussion by markder** [microsoft-search/pnp-modern-search#3767] [Q&A] [answered]
I am trying to show more information on the Contact portion of the persona hover card from AD. (custom attributes/extension attributes) I am able to do this on the built in Microsoft hover card on Outlook but it's a bit different for this project. Can someone tell me where in the source I should be looking to make these changes? Or should I be making changes in the results template?

**Comment by patrikhellgren** [comment]
Not sure if you already have, but you need to enable Microsoft Graph Toolkit to accomplish this.
Here is an example on how you could show a custom property in the additional details of the people card and this method can be used in layouts and in column Handlebars expressions:

```HTML
  <mgt-person
      person-query="{{getUserEmail @item.EditorOWSUSER}}"
      view="oneline"
      show-presence="true"
      person-card="hover"
  >
    <template data-type="person-card">
      <mgt-person-card inherit-details>
        <template data-type="additional-details">
          <mgt-get resource="/users/\{{ person.id }}?$select=extension_e7182bc3462d439f940ec10d277071f6_CustomerID" version="v1.0">
            <template>
              <div>
                <div data-if="extension_e7182bc3462d439f940ec10d277071f6_CustomerID">
                  <h4>Customer ID:</h4>
                    \{{ extension_e7182bc3462d439f940ec10d277071f6_CustomerID }}
                </div>
              </div>
            </template>
          </mgt-get>
        </template>
      </mgt-person-card>
    </template>
  </mgt-person>
```

You could use this method for any other supported section of the people card also, see the Microsoft Graph Toolkit documentation for more information.

I also thought this could be useful information for others so I put together a quick post on this topic with a little more information here [https://www.patrikhellgren.com/blog/how-to-customize-the-people-hover-card-in-pnp-modern-search/](https://www.patrikhellgren.com/blog/how-to-customize-the-people-hover-card-in-pnp-modern-search/).

**Reply by markder** [reply]
Thanks Patrik. This is very thorough. When trying to use this, I'm getting a null value in the Editor regardless of whether the user has an entry for that field. Following your example, I replaced "extension_e7182bc3462d439f940ec10d277071f6_CustomerID" with extensionAttribute2. I assume it's because I'm not getting the full extension name like your example property.  If that's the case, can you tell me how to find this?

![image](https://github.com/microsoft-search/pnp-modern-search/assets/2627563/e98c9b59-dd7e-4c2a-87bc-d5b583a683ad)

**Reply by patrikhellgren** [reply, marked as answer]
My example was based on the assumption that you were listing documents or list items. When listing people like in the image you cannot use EditorOWSUSER since it is null for all people returned.
Use UserName instead and modify the Handlebars expression like this and you should see the avatar and name rendered at least for the internal users. As for the extensionAttribute2 I assume you are talking about Exchange extension attribute synchronized from on-prem, then you would need to use syntax similar to the one in the below example. I don't have this available in my environments so I cannot test:

```HTML
  <mgt-person
      person-query="{{@item.UserName}}"
      view="oneline"
      show-presence="true"
      person-card="hover"
  >
    <template data-type="person-card">
      <mgt-person-card inherit-details>
        <template data-type="additional-details">
          <mgt-get resource="/users/\{{ person.id }}/onPremisesExtensionAttributes" version="v1.0">
            <template>
              <div>
                <div data-if="extensionAttribute2">
                  <h4>Customer ID:</h4>
                    \{{ extensionAttribute2 }}
                </div>
              </div>
            </template>
          </mgt-get>
        </template>
      </mgt-person-card>
    </template>
  </mgt-person>
```

To troubleshoot this if it does not display the extensionAttribute, use DevTools in the browser and have a look at the Graph call in the Network tab to see what the call returns.
You could also try the call in the [Graph Explorer](https://developer.microsoft.com/en-us/graph/graph-explorer) to see the returned data structure.

**Reply by markder** [reply]
Thank you so much, this worked.

**Reply by markder** [reply]
With this information, I was able to modify the People layout results template to get the desired hover card. I found the <template data-type="person card'> and added to it the "additional-details" part that you show here. 
![image](https://github.com/microsoft-search/pnp-modern-search/assets/2627563/2077955a-88ec-4be2-bf8f-c4e33f533567)

**Comment by thekaranjit** [comment]
Hello @patrikhellgren  @markder  Can you please share the code you used ? also any idea how can i hide or show details like, Organization, JobTitle & Department . 

I also want to hide below features of Hover card (Highlighted in the screenshot)
<img width="297" height="376" alt="Screenshot 2026-04-04 025607" src="https://github.com/user-attachments/assets/9405c94a-bb9a-4307-b22f-ec79cc659156" />

**Reply by patrikhellgren** [reply]
I wrote a blog post explaining this a bit more which you can find here [https://www.patrikhellgren.com/blog/how-to-customize-the-people-hover-card-in-pnp-modern-search/](https://www.patrikhellgren.com/blog/how-to-customize-the-people-hover-card-in-pnp-modern-search/).

Here is the documentation for the ```mgt-person-card``` component [https://learn.microsoft.com/en-us/graph/toolkit/components/person-card?tabs=html](https://learn.microsoft.com/en-us/graph/toolkit/components/person-card?tabs=html) where you can see what is possible with the component.

If I remember correctly you cannot hide those sections without using javascript and you cannot do that in the layouts without developing your own [custom web components](https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/). Maybe you could try to hide them using css in the layout.
